cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2460,15 +2466,9 @@ macro(build_grpc)
       -DgRPC_RE2_PROVIDER=package
       -DgRPC_SSL_PROVIDER=package
       -DgRPC_ZLIB_PROVIDER=package
-      -DgRPC_MSVC_STATIC_RUNTIME=${ARROW_USE_STATIC_CRT}
       -DCMAKE_INSTALL_PREFIX=${GRPC_PREFIX}
       -DCMAKE_INSTALL_LIBDIR=lib
       -DBUILD_SHARED_LIBS=OFF)
-  if(RE2_VENDORED)
-    list(APPEND GRPC_CMAKE_ARGS -Dre2_ROOT=${RE2_PREFIX}
```

Comment:
We can't use `<PackageName>_ROOT` with CMake 3.11 or older.
(This is not related to MSVC.)